packages/xstate-solid/src/useActor.ts

```diff
@@ -0,0 +1,71 @@
+import type { ActorRef, Event, EventObject, Sender } from 'xstate';
+import type { Accessor } from 'solid-js';
+import { createEffect, createMemo, on, onCleanup } from 'solid-js';
+import { State } from 'xstate';
+import { deriveServiceState } from './deriveServiceState';
+import { createImmutable } from './createImmutable';
+
+type EmittedFromActorRef<
+  TActor extends ActorRef<any>
+> = TActor extends ActorRef<any, infer TEmitted> ? TEmitted : never;
+
+const noop = () => {
+  /* ... */
+};
+
+// Only spread actor snapshot if it is a xstate state class
+const spreadIfStateInstance = <T>(value: T) =>
+  value instanceof State ? { ...value } : value;
```

Comment:
in `deriveServiceState` you are using `isStateLike` and I think that I would prefer using that strategy here too. `instanceof` can be very problematic in `node_modules`, an instance created by one copy of XState wont pass the check testing against the class from the second copy of XState